common/src/main/java/org/apache/uniffle/common/compression/Lz4Compressor.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.uniffle.common.compression;
+
+import net.jpountz.lz4.LZ4Compressor;
+import net.jpountz.lz4.LZ4Factory;
+
+public class Lz4Compressor implements Compressor {
+
+  private LZ4Compressor compressor;
+
+  public Lz4Compressor() {
+    this.compressor = LZ4Factory.fastestInstance().fastCompressor();
```

Comment:
One question:
Is it a thread safe? 
In our origin implement, we will use one fastestInstance in one thread.
In this pr's implement, we will share one fastestInstance in all threads.

Reply:
Nice catch. 

After reviewing the Spark related code, I think we should make `LZ4Factory.fastestInstance()` shareable.

Reply:
Updated.